src/python/pants/backend/native/register.py

```diff
@@ -35,6 +38,7 @@ def build_file_aliases():
 def register_goals():
   # FIXME(#5962): register these under the 'compile' goal when we eliminate the product transitive
   # dependency from export -> compile.
+  task(name='native-third-party-prep', action=NativeThirdPartyPrep).install('native-compile')
```

Comment:
I am confused here. It is piggybacking off the new `native-compile` goal created by Danny, right? And you would never run `native-third-party-prep` without running the compile and shared library tasks.

Reply:
Unless a goal is meant to run from the commandline, goal names (and goals themselves, really) aren't useful. The reason to collect tasks together into a goal is because you want a human to be able to use that goal name on the CLI and force all tasks in the goal to run. Maybe that's the intent with `native-compile`, but most likely what you'd want them to run is `./pants compile`, which will have a dependency on `native-compile`, and transitively on dep resolution and etc.